### PR TITLE
Replace culture-sensitive operations with ordinal operations

### PR DESCRIPTION
**Pull request by meziantou**
**Details of Changes**

- Use culture-insensitive comparisons for correctness. Depending on the culture `"Test".StartsWith(".", StringComparison.CurrentCulture)` could be true, which can be problematic when dealing with file extensions. Using `Ordinal`/`OrdinalIgnoreCase` also slightly increases the performance
- Replace `StringComparison.InvariantCulture` with `Ordinal`

For reference: https://www.meziantou.net/string-comparisons-are-harder-than-it-seems.htm

### CODE REVIEW

File: Files/Filesystem/StorageEnumerators/Win32StorageEnumerator.cs [hunk 1]
```diff
@@ -279,7 +279,7 @@ CancellationToken cancellationToken
                     if (status == AppServiceResponseStatus.Success
                         && response.ContainsKey("TargetPath"))
                     {
-                        var isUrl = findData.cFileName.EndsWith(".url");
+                        var isUrl = findData.cFileName.EndsWith(".url", StringComparison.Ordinal);
```
Comment (d2dyno1) [inline Files/Filesystem/StorageEnumerators/Win32StorageEnumerator.cs:282]: `OrdinalIgnoreCase`

**Review by d2dyno1 — CHANGES_REQUESTED** [review]
File extension comparasions can be compared with `OrdinalIgnoreCase` since an extension like e.g. ".exe" and ".EXE" work in the same manner

File: Files.Launcher/Helpers/ShellNewMenuHelper.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public static class ShellNewMenuHelper
         public static async Task<List<ShellNewEntry>> GetNewContextMenuEntries()
         {
             var newMenuItems = new List<ShellNewEntry>();
-            foreach (var keyName in Registry.ClassesRoot.GetSubKeyNames().Where(x => x.StartsWith(".") && !new string[] { ShellLibraryItem.EXTENSION, ".url", ".lnk" }.Contains(x)))
+            foreach (var keyName in Registry.ClassesRoot.GetSubKeyNames().Where(x => x.StartsWith('.') && !new string[] { ShellLibraryItem.EXTENSION, ".url", ".lnk" }.Contains(x)))
```
Comment (meziantou) [inline Files.Launcher/Helpers/ShellNewMenuHelper.cs:19]: Based on you comment, should I also change this one?

```suggestion
            foreach (var keyName in Registry.ClassesRoot.GetSubKeyNames().Where(x => x.StartsWith('.') && !new string[] { ShellLibraryItem.EXTENSION, ".url", ".lnk" }.Contains(x, StringComparer.OrdinalIgnoreCase)))
```

Comment (meziantou) [inline Files.Launcher/Helpers/ShellNewMenuHelper.cs:19]: Should this one be OrdinalIgnoreCase?

```suggestion
            foreach (var keyName in Registry.ClassesRoot.GetSubKeyNames().Where(x => x.StartsWith('.') && !new string[] { ShellLibraryItem.EXTENSION, ".url", ".lnk" }.Contains(x, StringComparer.OrdinalIgnoreCase)))
```

Comment (d2dyno1) [reply Files.Launcher/Helpers/ShellNewMenuHelper.cs:19]: Yes, OridnalIgnoreCase should be there too :)

Comment (d2dyno1) [reply Files.Launcher/Helpers/ShellNewMenuHelper.cs:19]: Yes

**Comment by meziantou** [comment]
> File extension should be compared with OrdinalIgnoreCase since extensions like ".exe" and ".EXE" both work in the same manner

All locations have been updated 🙂

**Review by d2dyno1 — APPROVED** [review]
Good work, LGTM!

**Pull request merged**